pkgs/tools/admin/azure-cli/default.nix

```diff
@@ -16,7 +28,48 @@ let
   };
 
   # put packages that needs to be overridden in the py package scope
-  py = callPackage ./python-packages.nix { inherit src version; };
+  py = callPackage ./python-packages.nix { inherit src version withImmutableConfig; };
+
+  # Builder for Azure CLI extensions. Extensions are Python wheels that
+  # outside of nix would be fetched by the CLI itself from various sources.
+  mkAzExtension =
+    { pname
+    , version
+    , url
+    , sha256
+    , description
+    }: python3.pkgs.buildPythonPackage {
+      inherit pname version;
+      format = "wheel";
+      src = fetchurl { inherit url sha256; };
+      meta = with lib; {
+        inherit description;
+        homepage = "https://github.com/Azure/azure-cli-extensions";
+        changelog = "https://github.com/Azure/azure-cli-extensions/blob/main/src/${pname}/HISTORY.rst";
+        license = lib.licenses.mit;
+      };
+    };
+
+  extensions = import ./extensions-generated.nix { inherit mkAzExtension; };
+
+  extensionDir = stdenvNoCC.mkDerivation {
+    name = "azure-cli-extensions";
+    phases = [ "installPhase" ];
```

Comment:
We wouldn't overwrite phases like this. It's there a particular reason they should be disabled?

Reply:
Ture, `dontUnpack = true` should work as well. Or should we just use `runCommand`? We don't really need anything from `mkDerivation` here, do we?